renpy.el

```diff
@@ -1981,7 +1987,85 @@ Use enclosing keywords to find image name symbol bounds.  Return a (BEG
 			 ;; TODO: Once we know the precise context, other
 			 ;; completion engines should not apply.
 			 :exclusive 'no)))))
-
+
+;;;; Flymake
+
+(defvar renpy--flymake-regex-list
+  (list
+   ;; Example: path/to/scrip.rpy:111 Error message
+   (rx bol
+       (group (1+ (not (any ?: "\n"))))
+       ":"
+       (group (1+ digit))
+       " "
+       (group (0+ (not "\n")))
+       eol)
+   ;; Example: File \"path/to/file/script.rpy\", line 27: expected statement.
+   (rx "File "
+       "\""
+       (group (1+ (not "\"")) )
+       "\", line "
+       (group (1+ digit))
+       ": "
+       (group (0+ (not "\n")))
+       eol))
+  "Regexps used to catch renpy lint errors/warnings.")
+
+(defvar-local renpy--flymake-proc nil
+  "Flymake process for the current buffer.")
+
+(defun renpy--find-project-root ()
+  "Find the Ren'Py project root by looking for a \"game/\" directory.
+Return the full path to the \"game/\" directory if found, or nil if not."
+  (when-let (dir (locate-dominating-file default-directory "game"))
```

Comment:
This would be better as `and-let*` because `when-let` will be marked as obsolete in Emacs 31, and we are interested in the return value rather than a side-effect.
```diff
--- a/renpy.el
+++ b/renpy.el
@@ -2017,7 +2017,7 @@ defvar-local renpy--flymake-proc
 (defun renpy--find-project-game-directory ()
   "Find the Ren'Py project \"game/\" directory.
 Return the full path to the \"game/\" directory if found, or nil if not."
-  (when-let (dir (locate-dominating-file default-directory "game"))
+  (and-let* ((dir (locate-dominating-file default-directory "game")))
     (expand-file-name "game" dir)))
 
 (defun renpy--parse-lint-buffer (buffer)
```